src/main/java/seedu/address/model/volunteer/SkillNameContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,88 @@
+package seedu.address.model.volunteer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.skill.Skill;
+
+/**
+ * Tests that a {@code Volunteer}'s {@code Name} matches any of the keywords given.
+ */
+public class SkillNameContainsKeywordsPredicate implements Predicate<Volunteer> {
+    private final List<Name> names;
+    private final List<Skill> skills;
+
+    /**
+     * Constructor that takes in names and skills
+     * @param names list of names
+     * @param skills list of skills
+     */
+    public SkillNameContainsKeywordsPredicate(List<Name> names, List<Skill> skills) {
+        this.names = names;
+        this.skills = skills;
+    }
+
+    /**
+     * Constructor that only takes in names
+     * @param names list of names
+     */
+    public SkillNameContainsKeywordsPredicate(List<Name> names) {
+        this.names = names;
+        this.skills = new ArrayList<>();
+    }
+
+    @Override
+    public boolean test(Volunteer volunteer) {
+        if (names.isEmpty()) {
+            assert !skills.isEmpty() : "the check for both skills and names not being empty is done in"
+                    + " parse method of VolunteerFindCommandParser class";
+            return skills.stream()
+                    .allMatch(skill ->
```

Comment:
why over here skills need `allMatch` but at the bottom is `anyMatch`?

Reply:
good pt, i shd add in comments. Cos if you have a command e.g. `vfind s/chef s/boxer`, then you want to find volunteers that are **both** chef and boxer. Whereas if you have command `vfind n/alice n/bob`, you want to find volunteers that names are **either** alice or bob

Reply:
ahhh ok interesting